action.yml

```
@@ -46,11 +46,35 @@ runs:
     - name: Download hadolint and make it available in path
       shell: bash
       run: |
-        echo "::debug::Downloading Hadolint ${{ inputs.version }}"
-        mkdir ${{ github.action_path }}/bin
-        curl -L -s -o ${{ github.action_path }}/bin/hadolint \
-          "https://github.com/hadolint/hadolint/releases/download/v${{ inputs.version }}/hadolint-Linux-x86_64"
-        chmod +x ${{ github.action_path }}/bin/hadolint
+        download_hadolint=false
+        # Check if hadolint exists
+        if [ ! -f "${{ github.action_path }}/bin/hadolint" ]; then
+          echo "::debug::Hadolint binary not found"
+          mkdir -p ${{ github.action_path }}/bin
```

Lets skip `-p` here; I kind of want it to fail if we reach this code with the new logic since its unexpected.

Also, we can move this to the place we download it (no need to create it in the path of not having to download anythin)

```suggestion
          mkdir ${{ github.action_path }}/bin
```

---

I'll move it as suggested but keep `-p` option to be compatible with the case of having to switch from one version to another. If we don't keep it, moving this line of code to the download section will result in triggering the same errors as before (since we try to create an existing path)